### PR TITLE
Integration of Calvalus enhancements into master

### PR DESCRIPTION
**Pull request by marpet**

### CODE REVIEW

File: idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/cloudshadow/S2IdepixPostCloudShadowOp.java [hunk 1]
```diff
@@ -181,7 +186,11 @@ public void initialize() throws OperatorException {
     }
 
     private float getRasterNodeValueAtCenter(RasterDataNode var, int width, int height) {
```
[the remark this answers is not in the document]
Comment (martin-boettcher) [reply idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/cloudshadow/S2IdepixPostCloudShadowOp.java:188]: same as above

File: idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/mountainshadow/SlopeAspectOrientationOp.java [hunk 2]
```diff
@@ -76,7 +81,7 @@ public void initialize() throws OperatorException {
                 throw new OperatorException("Could not retrieve spatial resolution from Geo-coding");
             }
         } else {
-            throw new OperatorException("Could not retrieve spatial resolution from Geo-coding");
+            spatialResolution = S2IdepixUtils.determineResolution(sourceProduct);
```
[the remark this answers is not in the document]
Comment (martin-boettcher) [reply idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/mountainshadow/SlopeAspectOrientationOp.java:84]: Fine with me not to integrate this. - I need it in the calvalus branch nevertheless. The input is a mosaic of S2 MSI resampled to 60 m. It is provided as NetCDF4-BEAM. And it has a proper crs variable defining the CRS. If I remember right, SNAP unfortunately does not recognise the provided CRS. I had to add lat and lon 2-D variables to make it work. Maybe, in the long run reading products with CRS could be better supported. Example inputs are in calvalus-production2:/hroc/calvalus/projects/hroc/l1-mosaic-v01/4nws/2022-11/29/A-094-31U/MSIL1C-60m-4nws-31U-A-094-20221129-v01.3.nc .

File: idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/cloudshadow/S2IdepixPreCloudShadowOp.java [hunk 3]
```diff
@@ -168,7 +173,11 @@ private float[] getSamples(RasterDataNode rasterDataNode, Rectangle rectangle) {
     }
 
     private float getRasterNodeValueAtCenter(RasterDataNode var, int width, int height) {
```
[the remark this answers is not in the document]
Comment (marpet) [reply idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/cloudshadow/S2IdepixPreCloudShadowOp.java:162]: Yes. true

File: idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/mountainshadow/SlopeAspectOrientationOp.java [hunk 4]
```diff
@@ -110,12 +115,24 @@ public void computeTileStack(Map<Band, Tile> targetTiles, Rectangle targetRectan
...
-                                                                          sourceLongitudes);
+        if (getSourceProduct().getSceneGeoCoding() instanceof CrsGeoCoding) {
+            ((CrsGeoCoding) getSourceProduct().getSceneGeoCoding()).
+                    getPixels((int) sourceRectangle.getMinX(),
+                              (int) sourceRectangle.getMinY(),
+                              (int) sourceRectangle.getWidth(),
+                              (int) sourceRectangle.getHeight(),
+                              sourceLatitudes,
+                              sourceLongitudes);
+        } else {
```
[the remark this answers is not in the document]
Comment (martin-boettcher) [reply idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/operators/mountainshadow/SlopeAspectOrientationOp.java:126]: fine with me - same comment as with previous

File: idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/S2IdepixClassificationOp.java [hunk 5]
```diff
@@ -82,6 +82,9 @@ public class S2IdepixClassificationOp extends Operator {
             description = " Threshold CL_THRESH")
     private double clThresh;
 
+    @Parameter(defaultValue = "true", label = " Classify invalid pixels as land/water")
+    private boolean classifyInvalid;
```
[the remark this answers is not in the document]
Comment (martin-boettcher) [reply idepix-s2msi/src/main/java/org/esa/snap/idepix/s2msi/S2IdepixClassificationOp.java]: I have removed it.

**Pull request merged**